code/android-core-library/src/phone/java/com/adobe/marketing/mobile/MobileCore.java

```diff
@@ -846,195 +729,201 @@ public static void lifecycleStart(final Map<String, String> additionalContextDat
      * This method should be called from the Activity onPause method.
      */
     public static void lifecyclePause() {
-        if (eventHub == null) {
-            Log.debug(LOG_TAG, "Failed to pause lifecycle session (%s)", NULL_CONTEXT_MESSAGE);
-            return;
-        }
-
-        EventData eventData = new EventData();
-        eventData.putString(CoreConstants.EventDataKeys.Lifecycle.LIFECYCLE_ACTION_KEY,
+        Map<String, Object> eventData = new HashMap<>();
+        eventData.put(CoreConstants.EventDataKeys.Lifecycle.LIFECYCLE_ACTION_KEY,
                 CoreConstants.EventDataKeys.Lifecycle.LIFECYCLE_PAUSE);
 
-        Event event = new Event.Builder("LifecyclePause", EventType.GENERIC_LIFECYLE, EventSource.REQUEST_CONTENT)
-                .setData(eventData)
+        Event event = new Event.Builder("LifecyclePause", EventType.TYPE_GENERIC_LIFECYLE, EventSource.TYPE_REQUEST_CONTENT)
+                .setEventData(eventData)
                 .build();
-
-        eventHub.dispatch(event);
+        dispatchEvent(event);
     }
 
+    // ========================================================
+    // Track methods
+    // ========================================================
+
     /**
-     * Collect PII data. Although using this call enables collection of PII data, the SDK does not
-     * automatically send the data to any Adobe endpoint.
+     * This method dispatches an Analytics track {@code action} event
+     * <p>
+     * Actions represent events that occur in your application that you want to measure; the corresponding metrics will
+     * be incremented each time the event occurs. For example, you may want to track when an user click on the login
+     * button or a certain article was viewed.
+     * <p>
      *
-     * @param data the map containing the PII data to be collected
+     * @param action      {@code String} containing the name of the action to track
+     * @param contextData {@code Map<String, String>} containing context data to attach on this hit
      */
-    public static void collectPii(final Map<String, String> data) {
-        if (eventHub == null) {
-            Log.debug(LOG_TAG, "Failed to collect PII (%s)", NULL_CONTEXT_MESSAGE);
-            return;
-        }
-
-        if (data == null || data.isEmpty()) {
-            Log.debug(LOG_TAG, "Could not trigger PII, the data is null or empty.");
-            return;
-        }
-
-        final EventData eventData = new EventData()
-                .putStringMap(CoreConstants.EventDataKeys.Signal.SIGNAL_CONTEXT_DATA, data);
-        eventHub.dispatch(new Event.Builder("CollectPII", EventType.GENERIC_PII,
-                EventSource.REQUEST_CONTENT).setData(eventData).build());
-        Log.trace(LOG_TAG, "Collect PII event was sent");
+    public static void trackAction(final String action, final Map<String, String> contextData) {
```

Comment:
null/empty check?

Reply:
I will check with the Analytics squad and update trackState/trackAction later